File: 12-doubly-linked-list/01-insert-at-beginning-of-dll.py
from DoublyLinkedList import ListNode


def insert_at_beginning_of_doubly_linked_list(head, x):
  node = ListNode(x, None, head)
  if head:
    head.prev = node
  return node


head = ListNode.create([10, 15, 20])
# print(head)
res = insert_at_beginning_of_doubly_linked_list(head, 5)
print(res)

File: 12-doubly-linked-list/01-insert-at-beginning-of-dll/01-insert-at-beginning-of-dll.ts
import { ListNode } from '../../TypeScriptDataStructure/DoublyLinkedList/ListNode'

const insert_at_beginning_of_dll = (head: ListNode, key: number) => {
  const new_head = new ListNode(key, null, head)
  head.prev = new_head
  return new_head
}

const head = new ListNode(10, null, new ListNode(20))
head.next.prev = head

console.log(insert_at_beginning_of_dll(head, 30))